sql/hive/src/main/scala/org/apache/spark/sql/hive/client/package.scala

```diff
@@ -131,8 +131,16 @@ package object client {
         "org.pentaho:pentaho-aggdesigner-algorithm",
         "org.apache.hive:hive-vector-code-gen"))
 
+    // Since HIVE-14496, Hive.java uses calcite-core
+    case object v4_0 extends HiveVersion("4.0.0",
+      extraDeps = Seq("org.apache.derby:derby:10.14.2.0"),
+      exclusions = Seq("org.apache.calcite:calcite-druid",
+        "org.apache.curator:*",
+        "org.pentaho:pentaho-aggdesigner-algorithm",
```

Comment:
Hive 4.0-beta1 depends on calcite 1.25, since CALCITE-1474 (calcite 1.11), `org.pentaho:pentaho-aggdesigner`(available in Conjar, which already sunset) was upgraded to `net.hydromatic:aggdesigner`(available in Maven Central), thus I think this exclusion is invalid, simply remove it would be fine.